x/file_test.go

```diff
@@ -0,0 +1,124 @@
+package x
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+var dir string
+
+func TestWalkPathFunc_Positive(t *testing.T) {
+
+	dir := "../x"
+
+	filesGot := WalkPathFunc(dir, func(path string, isdir bool) bool {
+		return !isdir && strings.HasSuffix(path, ".go")
+	})
+
+	if len(filesGot) == 0 {
+		t.Errorf("Path doesn't exist: %s", dir)
+	}
+}
+
+func TestWalkPathFunc_Negative(t *testing.T) {
+
+	dir := "./x"
+
+	filesGot := WalkPathFunc(dir, func(path string, isdir bool) bool {
+		return !isdir && strings.HasSuffix(path, ".go")
+	})
+
+	if len(filesGot) != 0 {
+		t.Errorf("Path exist: %s, for this test try passing wrong path", dir)
+	}
+}
+
+func TestWriteGroupIdFile(t *testing.T) {
+
+	for i := 0; i < 3; i++ {
+		dir := filepath.Join("./For_WriteGroupIdFile", strconv.Itoa(i))
+		os.MkdirAll(dir, 0700)
+
+		switch i {
+		case 0:
+			got := WriteGroupIdFile(dir, uint32(i)).Error()
+			want := "ID written to group_id file must be a positive number"
+
+			if got != want {
+				t.Errorf("got : %s, wanted : %s", got, want)
+			}
+
+		case 1:
+			got := WriteGroupIdFile(dir, uint32(i))
+
+			if got != nil {
+				t.Errorf("got : %s, wanted : nil", got)
+			}
+
+		case 2:
+
+			WriteGroupIdFile(dir, uint32(i))
+
+		}
+	}
+
+	//Delete dirs
+	deleteDirs("./For_WriteGroupIdFile")
+
+}
+
+func TestReadGroupIdFile(t *testing.T) {
+
+	for i := 0; i < 3; i++ {
+
+		dir := filepath.Join("./For_WriteGroupIdFile", strconv.Itoa(i))
+		os.MkdirAll(dir, 0700)
+		groupFile := filepath.Join(dir, GroupIdFileName)
+		f, _ := os.OpenFile(groupFile, os.O_CREATE|os.O_WRONLY, 0600)
+		f.WriteString(strconv.Itoa(int(i)))
+		f.WriteString("\n")
+		f.Close()
+
+	}
+
+	for i := 0; i < 3; i++ {
+		switch i {
+		case 0:
+			dir = filepath.Join("./ABC")
+			group_id, err := ReadGroupIdFile(dir)
+
+			if group_id != 0 && err != nil {
```

Comment:
Since we expect that group_id should be 0, AND we expect that err should be nil, I recommend changing this to:
```
    assert.Equal(0, group_id)
    assert.True(err == nil)
```
Or, if you want to keep it closer to what you have:
```
    if group_id != 0 || err != nil {
        t.Error("Something Wrong!")
    }
```

Reply:
> assert.True(err == nil)

There's also assert.NoError(err), which is convention in other tests.

Reply:
Oh, Cool!  I wasn't aware of that assert, @matthewmcneely . Yes, that makes a lot of sense.

Reply:
I have added the following code:- 

assert.Equal(uint32(0), group_id, "If passed path is not a directory, ReadGroupIdFile() returns 0")
assert.Nil(err, "If passed path is not a directory, ReadGroupIdFile() returns nil as an error")